hibernate-core/src/main/java/org/hibernate/cfg/AnnotationBinder.java

```diff
@@ -760,7 +762,13 @@ else if ( InheritanceType.SINGLE_TABLE.equals( inheritanceState.getType() ) ) {
 		}
 
 		// try to find class level generators
-		HashMap<String, IdentifierGeneratorDefinition> classGenerators = buildGenerators( clazzToProcess, context );
+		HashMap<String, IdentifierGeneratorDefinition> classGenerators;
+		if ( context.getBuildingOptions().isIdGeneratorNameScopeGlobal() ) {
```

Comment:
Here, there is no difference between the if and else branches. Did you have something in mind and forgot to add it?

Reply:
good catch, Thanks @vladmihalcea